fe/fe-core/src/main/java/com/starrocks/http/rest/LoadAction.java

```diff
@@ -82,6 +85,39 @@ public static void registerAction(ActionController controller) throws IllegalArg
                 new LoadAction(controller));
     }
 
+    public static List<Long> selectNodes(String warehouseName) throws DdlException {
+        // Choose a backend sequentially, or choose a cn in shared_data mode
+        List<Long> nodeIds = new ArrayList<>();
+        if (RunMode.isSharedDataMode()) {
+            List<Long> computeIds = GlobalStateMgr.getCurrentState().getWarehouseMgr().getAllComputeNodeIds(warehouseName);
+            for (long nodeId : computeIds) {
+                ComputeNode node = GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getBackendOrComputeNode(nodeId);
+                if (node != null && node.isAvailable()) {
+                    nodeIds.add(nodeId);
+                }
+            }
+        } else {
+            SystemInfoService systemInfoService = GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo();
+            nodeIds = systemInfoService.getNodeSelector()
```

Comment:
this may have side effect to the original `seqChooseBackendIds()` behavior. there is an internal counter to record the sequence of the selection, here `allAvailableBackends().size()` passed down to the caller, which advanced the counter by the available size, and only choose the first one for load, and next time this interface called, it will be the same backend node returned from the seqChoose.

Reply:
updated